lib/mixins/acroform.js

```diff
@@ -0,0 +1,317 @@
+        fnFormat = `AFSpecial_Format`;
+        params = FORMAT_SPECIAL[opts.format.type]
+      } else {
+        let format = opts.format.type.charAt(0).toUpperCase() + opts.format.type.slice(1);
+        fnKeystroke = `AF${format}_Keystroke`;
+        fnFormat = `AF${format}_Format`;
+        params = this._formatParamsAsString(opts.format.params);
+        if (opts.format.type === 'date') {
+          fnKeystroke += 'Ex';
+        }
+      }
+      opts.AA = opts.AA ? opts.AA : {};
+      opts.AA.K = {
+        S: "JavaScript",
+        JS: new String(`${fnKeystroke}(${params});`)
+      };
+      opts.AA.F = {
+        S: "JavaScript",
+        JS: new String(`${fnFormat}(${params});`)
+      };
+    }
+    delete opts.format;
+    return opts;
+  },
+
+  _formatParamsAsString (params) {
+    if (Array.isArray(params)) {
+      return JSON.stringify(params).replace('[', '').replace(']', '')
+    }
+    return String(params);
+  },
+
+  _resolveColors (opts) {
+    let color = this._normalizeColor(opts.backgroundColor);
+    if (color) {
+      if (!opts.MK) {
+        opts.MK = {};
+      }
+      opts.MK.BG = color;
+    }
+    color = this._normalizeColor(opts.borderColor);
+    if (color) {
+      if (!opts.MK) {
+        opts.MK = {};
+      }
+      opts.MK.BC = color;
+    }
+    delete opts.backgroundColor;
+    delete opts.borderColor;
+    return opts;
+  },
+
+  _resolveFlags (options) {
+    let result = 0;
+    Object.keys(options).forEach(key => {
+      if (FIELD_FLAGS[key]) {
+        result |= FIELD_FLAGS[key]
+        delete options[key]
+      }
+    });
+    if (result !== 0) {
+      options.Ff = options.Ff ? options.Ff : 0;
+      options.Ff |= result;
+    }
+    return options;
+  },
+
+  _resolveJustify (options) {
+    let result = 0;
+    if (options.align !== undefined) {
+      if (typeof FIELD_JUSTIFY[options.align] === 'number') {
+        result = FIELD_JUSTIFY[options.align];
+      }
+      delete options.align;
+    }
+    if (result !== 0) {
+      options.Q = result;   // default
+    }
+    return options;
+  },
+
+  _resolveFont (options) {
+    // add current font to document-level AcroForm dict if necessary
+    if (this._acroform.fonts[this._font.id] == null) {
+      this._acroform.fonts[this._font.id] = this._font.ref();
+    }
+
+    // add current font to field's resource dict (RD) if not the default acroform font
+    if (this._acroform.defaultFont !== this._font.name) {
+      options.DR = { Font: {} };
+      options.DR.Font[this._font.id] = this._font.ref();
+    }
+    return options;
+  },
+
+  _resolveStrings (options) {
+    if (Array.isArray(options.Opt)) {
+      let arr = [];
+      for (let idx = 0; idx < options.Opt.length; idx++) {
+        if (typeof options.Opt[idx] === 'string') {
+          arr.push(new String(options.Opt[idx]))
+        } else {
+          arr.push(options.Opt[idx])
+        }
+      }
+      options.Opt = arr;
+    }
+    ['V', 'DV'].forEach(key => {
```

Comment:
Should we have more descriptive names for these fields and the below?

Reply:
I've made them value and defaultValue and documented them.